pkg/gsclient/cluster.go

```diff
@@ -43,20 +37,18 @@ func (c *Client) DeleteCluster(ctx context.Context, clusterID string) error {
 		return microerror.Mask(err)
 	}
 
-	// TODO: extract and structure all these hardcoded values
-	output, err := c.runWithGsctl(ctx, "--output=json", "delete", "cluster", clusterID)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
 	var response DeletionResponse
-	err = json.Unmarshal(output.Bytes(), &response)
+	// TODO: extract and structure all these hardcoded values
```

Comment:
Any idea how we should do this?

Reply:
I haven't thought about it yet. This todo was there before and I didn't want to overload the PR with too many unrelated refactorings :)

Reply:
I think it would be cleaner if we imported `gsctl` as a library rather than calling the binary. Otherwise, we should probably create a function for each command that accepts the dynamic arguments and returns an array of arguments we pass in to `runWithGsctl`. That makes it testable and defines an interface for each command. Should we make a todo list in our board for these refactorings (including using Tekton results from above)?